File: junit/se/fredsberg/game/dice/zombie/ZombieDieTest.java
package se.fredsberg.game.dice.zombie;

import static org.junit.Assert.*;

import java.awt.Color;

import org.junit.Test;

public class ZombieDieTest {

    @Test
    public void redZombieDie_properties() {
        ZombieDie red = new ZombieDieFactory().redZombieDie();
        assertEquals(Color.RED, red.getColor());
        assertZombieDieSides(red, 1, 3, 2);
    }

    @Test
    public void yellowZombieDie_properties() {
        ZombieDie yellow = new ZombieDieFactory().yellowZombieDie();
        assertEquals(Color.YELLOW, yellow.getColor());
        assertZombieDieSides(yellow, 2, 2, 2);
    }

    @Test
    public void greenZombieDie_properties() {
        ZombieDie green = new ZombieDieFactory().greenZombieDie();
        assertEquals(Color.GREEN, green.getColor());
        assertZombieDieSides(green, 3, 1, 2);
    }

    private void assertZombieDieSides(ZombieDie die, int brains, int shootguns, int footprints) {
        int sides = shootguns + footprints + brains;
        assertEquals(sides + " sides", sides, die.getSize());
        assertEquals(shootguns + " shootguns", shootguns, countValues(die, ZombieValue.SHOOTGUN));
        assertEquals(footprints + " footprints", footprints, countValues(die, ZombieValue.FOOTPRINTS));
        assertEquals(brains + " brains", brains, countValues(die, ZombieValue.BRAIN));
    }

    private int countValues(ZombieDie die, ZombieValue value) {
        int result = 0;
        for (ZombieValue side : die.sides()) {
            if (side == value) {
                result++;
            }
        }
        return result;
    }
}

File: jstests/ZombieDieTest.js
var assert = require('assert');
var ZombieDieFactory = require('../jssrc/ZombieDieFactory.js');
var ZombieValue = require('../jssrc/ZombieValue.js');
var Color = require('../jssrc/Color.js');

module.exports = {

RedZombieDie_properties: function() {
	var red = new ZombieDieFactory().redZombieDie();
	assert.equal(new Color().colors.RED, red.getColor());
	assert.equal(6, red.getSize());
	var sides = red.getSides();
    var z = new ZombieValue();
	assert.equal(z.values.BRAIN, sides[0]);
	assert.equal(z.values.FOOTPRINTS, sides[1]);
	assert.equal(z.values.FOOTPRINTS, sides[2]);
	assert.equal(z.values.SHOOTGUN, sides[3]);
	assert.equal(z.values.SHOOTGUN, sides[4]);
	assert.equal(z.values.SHOOTGUN, sides[5]);
},

YellowZombieDie_properties: function() {
	var yellow = new ZombieDieFactory().yellowZombieDie();
	assert.equal(new Color().colors.YELLOW, yellow.getColor());
	assert.equal(6, yellow.getSize());
	var sides = yellow.getSides();
    var z = new ZombieValue();
	assert.equal(z.values.BRAIN, sides[0]);
	assert.equal(z.values.BRAIN, sides[1]);
	assert.equal(z.values.FOOTPRINTS, sides[2]);
	assert.equal(z.values.FOOTPRINTS, sides[3]);
	assert.equal(z.values.SHOOTGUN, sides[4]);
	assert.equal(z.values.SHOOTGUN, sides[5]);
},

GreenZombieDie_properties: function() {
	var green = new ZombieDieFactory().greenZombieDie();
	assert.equal(new Color().colors.GREEN, green.getColor());
	assert.equal(6, green.getSize());
	var sides = green.getSides();
    var z = new ZombieValue();
	assert.equal(z.values.BRAIN, sides[0]);
	assert.equal(z.values.BRAIN, sides[1]);
	assert.equal(z.values.BRAIN, sides[2]);
	assert.equal(z.values.FOOTPRINTS, sides[3]);
	assert.equal(z.values.FOOTPRINTS, sides[4]);
	assert.equal(z.values.SHOOTGUN, sides[5]);
}
};
